webapp/src/main/java/com/box/l10n/mojito/service/image/ImageServiceConfiguration.java

```diff
@@ -0,0 +1,70 @@
+package com.box.l10n.mojito.service.image;
+
+import com.box.l10n.mojito.service.blobstorage.s3.S3BlobStorage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+@Configuration
+public class ImageServiceConfiguration {
+
+    @ConditionalOnProperty(value = "l10n.image-service.storage.type", havingValue = "s3Fallback")
+    static class S3FallbackImageServiceConfiguration {
+
+        @Autowired
+        ImageRepository imageRepository;
+
+        @Autowired
+        S3BlobStorage s3BlobStorage;
+
+        @Bean("databaseImageService")
```

Comment:
I'm not 100% sure but on top of my mind `@Bean("databaseImageService")` defines the bean name (and not qualifier). If not provided it uses the method names. So, here it is probably redundant since it is the same as the method name. 

I think it is either 
- `@Bean @Qualifier("databaseImageService")` with `@Qualifier("databaseImageService")` in the method definition. 
- Or if using bean names `@Bean("databaseImageService")` (which with this method name could be @Bean only) it matches on the method param names `databaseImageService` and the qualifier is not used. 

If my assumption are correct, to make it easier to read and less brittle to code changes, it is better to use @Qualifier